lib/src/client.dart

```diff
@@ -927,58 +927,66 @@ class Client extends MatrixApi {
-          client: this,
-          roomAccountData:
-              room.accountData?.asMap().map((k, v) => MapEntry(v.type, v)) ??
-                  <String, BasicRoomEvent>{},
-        );
+        _storeArchivedRoom(entry.key, entry.value);
+      }
+    }
+    return _archivedRooms;
+  }
 
-        final timeline = Timeline(
-            room: leftRoom,
-            chunk: TimelineChunk(
-                events: room.timeline?.events?.reversed
-                        .toList() // we display the event in the other sence
-                        .map((e) => Event.fromMatrixEvent(e, leftRoom))
-                        .toList() ??
-                    []));
-
-        leftRoom.prev_batch = room.timeline?.prevBatch;
-        room.state?.forEach((event) {
-          leftRoom.setState(Event.fromMatrixEvent(
-            event,
-            leftRoom,
-          ));
-        });
+  void _storeArchivedRoom(String id, LeftRoomUpdate update) {
+    final room = update;
+    final leftRoom = Room(
+      id: id,
+      membership: Membership.leave,
+      client: this,
+      roomAccountData:
+          room.accountData?.asMap().map((k, v) => MapEntry(v.type, v)) ??
+              <String, BasicRoomEvent>{},
+    );
 
-        room.timeline?.events?.forEach((event) {
-          leftRoom.setState(Event.fromMatrixEvent(
-            event,
-            leftRoom,
-          ));
-        });
+    final timeline = Timeline(
+        room: leftRoom,
+        chunk: TimelineChunk(
+            events: room.timeline?.events?.reversed
+                    .toList() // we display the event in the other sence
+                    .map((e) => Event.fromMatrixEvent(e, leftRoom))
+                    .toList() ??
+                []));
+
+    leftRoom.prev_batch = room.timeline?.prevBatch;
+    room.state?.forEach((event) {
+      leftRoom.setState(Event.fromMatrixEvent(
+        event,
+        leftRoom,
+      ));
+    });
 
-        for (var i = 0; i < timeline.events.length; i++) {
-          // Try to decrypt encrypted events but don't update the database.
-          if (leftRoom.encrypted && leftRoom.client.encryptionEnabled) {
-            if (timeline.events[i].type == EventTypes.Encrypted) {
-              timeline.events[i] =
-                  await leftRoom.client.encryption!.decryptRoomEvent(
-                leftRoom.id,
-                timeline.events[i],
-              );
-            }
-          }
-        }
+    room.timeline?.events?.forEach((event) {
+      leftRoom.setState(Event.fromMatrixEvent(
+        event,
+        leftRoom,
+      ));
+    });
 
-        _archivedRooms.add(ArchivedRoom(room: leftRoom, timeline: timeline));
+    for (var i = 0; i < timeline.events.length; i++) {
+      // Try to decrypt encrypted events but don't update the database.
+      if (leftRoom.encrypted && leftRoom.client.encryptionEnabled) {
+        if (timeline.events[i].type == EventTypes.Encrypted) {
+          unawaited(
```

Comment:
Can you add a comment for why this is unawaited? (is not a very obvious change for td)

Reply:
Hmmmm, I was unsure about that one. I didn't find it necessary to do this synchronously because of the number of changes (up to the `_handleRooms` method) but maybe yeah, might be better to await this. Correspondingly updated it.